salt/_runners/metalk8s_checks.py

```diff
@@ -0,0 +1,246 @@
+            minion for minion, minion_ret in ret.items() if minion_ret is not True
+        ],
+        "did not answered": list(set(minion_list) - set(ret.keys())),
+    }
+
+    for reason, minions_names in not_ready_minions.items():
+        if minions_names:
+            errors.append(
+                "Salt minions '{}' {}".format("', '".join(minions_names), reason)
+            )
+
+    return _handle_errors(errors, raises)
+
+
+def upgrade(dest_version, saltenv, raises=True):
+    """Check that we can start MetalK8s cluster upgrade
+
+    Args:
+        dest_version (string): Destination version for upgrade
+        saltenv (string): Salt environment that will be used for upgrade
+        raises (bool, optional): Whether or not this function should raise. Defaults to True.
+
+    Raises:
+        CheckError: If 'raises' is True and some upgrade condition are not met
+
+    Returns:
+        dict or True: An error message or True if everything OK
+    """
+    errors = []
+
+    # NOTE: We use `get_from_map` as we want also defaults from `map.jinja`
+    # to be merged
+    metalk8s_pillar = __salt__["salt.cmd"](
+        "metalk8s.get_from_map", "metalk8s", saltenv=saltenv, with_pillar=True
+    )
+
+    # When upgrading the saltenv should be the "metalk8s-<destination version>"
+    if saltenv != f"metalk8s-{dest_version}":
+        errors.append(
+            f"Invalid saltenv '{saltenv}' consider using 'metalk8s-{dest_version}'"
+        )
+
+    # Check that all nodes are in a supported upgrade path
+    # We only support upgrade from one minor version
+    dest = dest_version.split(".")
+    min_version = f"{dest[0]}.{int(dest[1]) - 1}.0"
+
+    for node_name, node_info in metalk8s_pillar["nodes"].items():
+        if (
+            __salt__["salt.cmd"]("pkg.version_cmp", min_version, node_info["version"])
+            == 1
+        ):
+            errors.append(
+                "Unable to upgrade from more than 1 minor version, Node "
+                f"{node_name} is in {node_info['version']} and you try to upgrade "
+                f"to {dest_version}"
+            )
+
+    nodes_ready = nodes(node_list=metalk8s_pillar["nodes"].keys(), raises=False)
+    if nodes_ready is not True:
+        errors.extend(nodes_ready["errors"])
+
+    minions_ready = minions(minion_list=metalk8s_pillar["nodes"].keys(), raises=False)
+    if minions_ready is not True:
+        errors.extend(minions_ready["errors"])
+
+    return _handle_errors(errors, raises)
+
+
+def downgrade(dest_version, saltenv, raises=True, bypass_disable=False):
+    """Check that we can start MetalK8s cluster downgrade
+
+    Args:
+        dest_version (string): Destination version for downgrade
+        saltenv (string): Salt environment that will be used for downgrade
+        raises (bool, optional): Whether or not this function should raise. Defaults to True.
+        bypass_disable (bool, optional): Whether or not bypass downgrade disable. Default to False.
```

Comment:
Yeah, well this explanation is not crystal clear and the option is not really self explanatory, I had to read the code to understand. :)
I would rather named it `force`, but you can keep it as is if you prefer, but mainly for the doc I would write something like:
```suggestion
        bypass_disable (bool, optional): Force downgrade if the downgrade path is not supported. Default to False.
```

Reply:
Right ok, since it's a "check" function I'm not sure about "force" but right I will update the docstring